src/seq/nucleotide.jl

```diff
@@ -119,16 +119,27 @@ end
 # represented with an N mask BitArray. If the ns[i] bit is set, then the
 # sequence may have any nucleotide at that position and it must be ignored.
 #
-# Sequences are immutable by convention. Low-level functions will mutate the
-# underlying data, any user-facing function that does so should have `unsafe` in
-# the name. Immutability allows two sequences to share the same underlying data,
```

Comment:
I don't think SubArray would be especially useful here. What we are doing is quite similar to SubArray, but for subsequences we need to be able to specify the range at the resolution of bits, since this is 2-bit encoded, whereas a SubArray would only let us specify ranges at best at the byte level.